### PR TITLE
chore: arrange github workflows by scope, add descriptive names

### PR DESCRIPTION
**Pull request by luke-biel**
When this get's approved we have to change `required` workflow names

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -6,10 +6,108 @@ on:
...
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+          architecture: 'x64'
+
+      - name: Cache cargo target dir
+        uses: actions/cache@v2.1.4
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
```
Comment (Kononnable) [inline .github/workflows/test.yml:97]:
```suggestion
        with:
          path: target
           key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
          restore-keys: ${{ runner.os }}-cargo-build-target-
```
Should work a bit better for build which changes single dependency.

Comment (Kononnable) [reply .github/workflows/test.yml:97]: And I agree, we should define an action for caching rust - we've got it used in few places and they're not synced.
But it's fine if this action is not be in scope for this PR

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -6,10 +6,108 @@ on:
       - "develop"
   pull_request:
 
+env:
```
Comment (Kononnable) [inline .github/workflows/test.yml:9]: We could probably limit those actions to not be started on rfc PRs/merges(ignore docs path or include just crates folder with `Cargo*` files.

**Comment by frondeus** [comment]
Besides what Paweł mentioned - LGTM

**Review by esavier — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -6,10 +6,109 @@ on:
...
+        with:
+          python-version: '3.x'
+          architecture: 'x64'
+
+      - name: Cache cargo target dir
+        uses: actions/cache@v2.1.4
+        with:
+          path: target
+           key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-build-target-
```
Comment (Kononnable) [inline .github/workflows/test.yml:98]: Not sure how that space got here. I think it was fine in proposed suggestion
```suggestion
          path: target
          key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
          restore-keys: ${{ runner.os }}-cargo-build-target-
```

Comment (Kononnable) [reply .github/workflows/test.yml:98]: Ok, I see it there too, sorry :)

File: .github/workflows/test.yml (hunk 2 above)
Comment (luke-biel) [reply .github/workflows/test.yml:14]: We have 2 `mandatory` actions, so I cannot (don't want to) remove running these actions on PRs, but I can omit running them on merges.

**Comment by luke-biel** [comment]
https://github.com/luke-biel/CommonDataLayer/pull/4 - to compare

**Review by frondeus — APPROVED** [review]

**Pull request merged**